numpy/lib/tests/test_format.py

```diff
@@ -952,17 +952,17 @@ def test_unicode_field_names():
             format.write_array(f, arr, version=None)
 
 
-@pytest.mark.parametrize('dt, fail', [
+@pytest.mark.parametrize('dt, fail, has_metadata', [
     (np.dtype({'names': ['a', 'b'], 'formats':  [float, np.dtype('S3',
-                 metadata={'some': 'stuff'})]}), True),
-    (np.dtype(int, metadata={'some': 'stuff'}), False),
-    (np.dtype([('subarray', (int, (2,)))], metadata={'some': 'stuff'}), False),
+                 metadata={'some': 'stuff'})]}), False, True),
+    (np.dtype(int, metadata={'some': 'stuff'}), False, False),
+    (np.dtype([('subarray', (int, (2,)))], metadata={'some': 'stuff'}), False, False),
     # recursive: metadata on the field of a dtype
     (np.dtype({'names': ['a', 'b'], 'formats': [
         float, np.dtype({'names': ['c'], 'formats': [np.dtype(int, metadata={})]})
-    ]}), False)
+    ]}), False, False)
```

Comment:
It seems we are now missing cases where `fail==True`?

Reply:
Correct. It does not fail anymore on np.load.  Should I remove option fail completely?